component/component.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2019 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package component
+
+import "context"
+
+// Component is either a receiver, exporter, processor or extension.
+type Component interface {
+	// Start tells the component to start. Host parameter can be used for communicating
+	// with the host after Start() has already returned. If error is returned by
+	// Start() then the collector startup will be aborted.
+	// If this is an exporter component it may prepare for exporting
+	// by connecting to the endpoint.
+	Start(host Host) error
+
+	// Shutdown is invoked during service shutdown.
+	Shutdown() error
+}
+
+type Host interface {
```

Comment:
nit: add a comment for `Host`, suggestion: `Host represents the entity that is hosting a Component. It is used to allow communication between the Component and its host.`

Reply:
Done.